src/main/java/org/mockbukkit/mockbukkit/MockBukkitExtension.java

```diff
@@ -93,35 +148,75 @@ public class MockBukkitExtension implements TestInstancePostProcessor, TestInsta
 
 	private final Logger logger = Logger.getLogger("MockBukkitExtension");
 	private final Set<Class<?>> serverSupportedTypes = Set.of(Server.class, ServerMock.class);
+	private final Set<Class<?>> playerSupportedTypes = Set.of(Player.class, PlayerMock.class);
+	private final Set<Class<?>> worldSupportedTypes = Set.of(World.class, WorldMock.class);
+	private final Set<Class<?>> pluginSupportedTypes = Set.of(Plugin.class, PluginMock.class);
+
+	private int playerCounter = 0;
+	private int worldCounter = 0;
+	private int pluginCounter = 0;
 
 	@Override
 	public void postProcessTestInstance(Object testInstance, ExtensionContext context) throws Exception
 	{
-		final ServerMock serverMock = MockBukkit.getOrCreateMock();
-		injectServerMockIntoFields(testInstance, context, serverMock);
+		injectMocksIntoFields(testInstance, context);
 	}
 
-	private void injectServerMockIntoFields(Object testInstance, ExtensionContext context, ServerMock serverMock) throws IllegalAccessException
+	private void injectMocksIntoFields(Object testInstance, @NotNull ExtensionContext context) throws IllegalAccessException
 	{
 		final Optional<Class<?>> classOptional = context.getTestClass();
 		if (classOptional.isEmpty())
 		{
 			return;
 		}
 
-		final List<Field> serverMockFields = FieldUtils.getAllFieldsList(classOptional.get())
+		final List<Field> allFields = FieldUtils.getAllFieldsList(classOptional.get())
 				.stream()
-				.filter(field -> serverSupportedTypes.contains(field.getType()))
 				.filter(field -> field.getAnnotation(MockBukkitInject.class) != null)
 				.toList();
 
-		for (final Field field : serverMockFields)
+		for (final Field field : allFields)
 		{
-			final String name = field.getName();
-			FieldUtils.writeDeclaredField(testInstance, name, serverMock, true);
+			final Object mockObject = createMockForField(field);
+			if (mockObject != null)
+			{
+				FieldUtils.writeField(field, testInstance, mockObject, true);
+			}
 		}
 	}
 
+	private @Nullable Object createMockForType(Class<?> type)
+	{
+		if (serverSupportedTypes.contains(type))
+		{
+			return MockBukkit.getOrCreateMock();
+		}
+		else if (playerSupportedTypes.contains(type))
+		{
+			final String playerName = "Player" + playerCounter++;
+			return MockBukkit.getOrCreateMock().addPlayer(playerName);
+		}
+		else if (worldSupportedTypes.contains(type))
+		{
+			final String worldName = "World" + worldCounter++;
```

Comment:
Would it otherwise be possible to require a name parameter string in the annotation, which you could use here?
You could have it be optional in the annotation, and then require it only for certain fields, throwing some kind of exception when the test starts. That could possibly clarify where this name comes from, or what do you think?

Reply:
Added, pure optional behaviour. In most of my tests, I don't care what a player is called, I just care I have a player object ;). But if it's necessary, you can name them however you want.